SYCL/DeviceLib/built-ins/ext_native_math.cpp

```diff
@@ -0,0 +1,90 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// OpenCL CPU driver does not support cl_khr_fp16 extension
+// UNSUPPORTED: cpu && opencl
```

Comment:
Presumably, the `float` and `float{2,3,4,8,16}` tests would still pass, though, right?  Rather than disabling this test for the CPU platform, why not add a runtime check to see if the device has `aspect::fp16`, and then only disable the `half` tests if the device doesn't have that aspect.

Reply:
Yes, it's fine to add `-fsycl-device-code-split=per_kernel` for now.  This won't be necessary once we implement the ["optional  device feature"](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md) support.  You could include a comment to that effect, so we know to remove the `-fsycl-device-code-split=per_kernel` later.

Reply:
Thanks for the info @gmlueck.

I included the modifications in the next commit.